packages/@expo/server/src/index.ts

```diff
@@ -79,55 +78,47 @@ export function createRequestHandler(
       }
       return import(filePath);
     },
-    logApiRouteExecutionError = (error: Error) => {
-      console.error(error);
-    },
-    handleApiRouteError = async (error: Error) => {
-      if ('statusCode' in error && typeof error.statusCode === 'number') {
-        return new Response(error.message, {
-          status: error.statusCode,
-          headers: { 'Content-Type': 'text/plain' },
-        });
-      }
-      return new Response('Internal server error', {
-        status: 500,
-        headers: { 'Content-Type': 'text/plain' },
-      });
+    handleRouteError = async (error: Error) => {
+      // In production the server should handle unexpected errors
+      throw error;
     },
   }: {
-    getHtml?: (request: Request, route: RouteInfo<RegExp>) => Promise<string | Response | null>;
-    getRoutesManifest?: (distFolder: string) => Promise<ExpoRoutesManifestV1<RegExp> | null>;
-    getApiRoute?: (route: RouteInfo<RegExp>) => Promise<any>;
+    getHtml?: (request: Request, route: Route) => Promise<string | Response | null>;
+    getRoutesManifest?: (distFolder: string) => Promise<Manifest | null>;
+    getApiRoute?: (route: Route) => Promise<any>;
     logApiRouteExecutionError?: (error: Error) => void;
-    handleApiRouteError?: (error: Error) => Promise<Response>;
+    handleRouteError?: (error: Error) => Promise<Response>;
```

Comment:
not for this PR / note to self: We could add some breaking changes to some of these signatures and make them easier to deal with. `distFolder` and `route` as the arguments seem conflicting with one another and not as easy to use as it could be.

Reply:
Yes, I agree, in the follow up (currently draft), the manifest getter has no arguments.

https://github.com/expo/expo/pull/38531/files#diff-8e21c18707f877e69c85d63d9738a0f458d279c02b62124ddb0791db43e816e5R18